pkg/runtime/controller.go

```diff
@@ -368,15 +386,25 @@ func (c *Controller) processRules(handlerConfig map[string]*cpb.Handler,
 
 var cleanupSleepTime = 500 * time.Millisecond
 
+// isFQN returns true if the name is fully qualified.
+// every resource name is defined by Key.String()
+// shortname.kind.namespace
+func isFQN(name string) bool {
+	return len(strings.Split(name, ".")) == 3
+}
+
 // processActions prunes actions that lack referential integrity and associate instances with
 // handlers that are later used to create new handlers.
 func (c *Controller) processActions(acts []*cpb.Action, handlerConfig map[string]*cpb.Handler,
-	instanceConfig map[string]*cpb.Instance, ht *handlerTable) map[adptTmpl.TemplateVariety]map[string]*Action {
+	instanceConfig map[string]*cpb.Instance, ht *handlerTable, namespace string) map[adptTmpl.TemplateVariety]map[string]*Action {
 
 	actions := make(map[adptTmpl.TemplateVariety]map[string]*Action)
 
 	for _, ic := range acts {
 		var hc *cpb.Handler
+		if !isFQN(ic.Handler) {
+			ic.Handler = ic.Handler + "." + namespace
+		}
```

Comment:
This section is not tested in the controller_test.go, right?  It's better to add tests for them.